src/consensus/tx_verify.cpp

```diff
@@ -262,7 +262,7 @@ bool Consensus::CheckTxInputs(const CTransaction& tx, CValidationState& state, c
     CAmount txfee_aux = nValueIn - value_out;
 
     if (tx.HasMWData()) {
-        // MW: TODO - Check peg-in maturity here?
+        libmw::node::CheckTxInputs(tx.m_mwtx.m_transaction, inputs.GetMWView(), nSpendHeight);
```

Comment:
This should be surrounded in a try catch. The litecoin codebase does not provide any exception safety guarantees, so just catch any exceptions and if an exception is thrown, handle it by return DoS:

```
        return state.DoS(100, false, REJECT_INVALID, "bad-txns-inputs-mweb", false,
                         strprintf("%s: MWEB inputs missing/immature", __func__));
```